src/sundials_matrix_wrapper.cpp

```diff
@@ -651,40 +652,38 @@ void SUNMatrixWrapper::transpose(SUNMatrixWrapper &C, const realtype alpha,
 
     // see https://github.com/DrTimothyAldenDavis/SuiteSparse/blob/master/CSparse/Source/cs_transpose.c
 
-    sunindextype aidx;
-    sunindextype cidx;
-    sunindextype acol;
-
     std::vector<sunindextype> w;
-
-    sunindextype ccol;
-    sunindextype crow;
-    sunindextype widx;
-
-    if (C.matrix_id()== SUNMATRIX_SPARSE) {
+    auto nrows = rows();
+    if (C_matrix_id == SUNMATRIX_SPARSE) {
         w = std::vector<sunindextype>(columns());
-        for (acol = 0; acol < rows(); acol++)                /* row counts */
-            for (aidx = get_indexptr(acol);
-                 aidx < get_indexptr(acol+1); aidx++) {
-                widx = (acol/blocksize)*blocksize + get_indexval(aidx) % blocksize;
-                w.at(widx)++;
-                assert(w[widx] <= rows());
+        for (sunindextype acol = 0; acol < nrows; acol++) { /* row counts */
+            auto next_indexptr = get_indexptr(acol+1);
+            for (sunindextype aidx = get_indexptr(acol);
+                 aidx < next_indexptr; aidx++) {
+                sunindextype widx = (acol/blocksize)*blocksize + get_indexval(aidx) % blocksize;
```

Comment:
@FFroehlich: I was wondering if all those `* blocksize / blocksize` are helpful here in any way

Reply:
Ah, all integers. Forget it.